fvm/handler/programs.go

```diff
@@ -102,58 +72,18 @@ func (h *ProgramsHandler) Get(location common.Location) (*interpreter.Program, b
 		return prog, ok
 	}
 
-	program, view, has := h.Programs.Get(address)
+	program, state, has := h.Programs.Get(address)
 	if has {
-		if view != nil { // handle view not set (ie. for non-address locations
-			// don't enforce limits while merging a cached view
-			enforceLimits := false
-			err := h.mergeState(view, enforceLimits)
-			if err != nil {
-				panic(fmt.Sprintf("merge error while getting program, panic: %s", err))
-			}
-		}
-		return program, true
-	}
-
-	parentState := h.masterState.State()
-	if len(h.viewsStack) > 0 {
-		parentState = h.viewsStack[len(h.viewsStack)-1].state
-	}
-
-	childState := parentState.NewChild()
-
-	h.viewsStack = append(h.viewsStack, stackEntry{
-		state:    childState,
-		location: location,
-	})
-
-	h.masterState.SetActiveState(childState)
-
-	return nil, false
-}
-
-func (h *ProgramsHandler) Cleanup() error {
-	stackLen := len(h.viewsStack)
-
-	if stackLen == 0 {
-		return nil
-	}
-
-	for i := stackLen - 1; i > 0; i-- {
-		entry := h.viewsStack[i]
-		err := h.viewsStack[i-1].state.MergeState(entry.state, h.masterState.EnforceInteractionLimits())
+		err := h.masterState.AttachAndCommitParseRestrictedToParent(state)
 		if err != nil {
-			return fmt.Errorf("cannot merge state while cleanup: %w", err)
+			panic(fmt.Sprintf("merge error while getting program, panic: %s", err))
 		}
+		return program, true
 	}
 
-	err := h.initialState.MergeState(h.viewsStack[0].state, h.masterState.EnforceInteractionLimits())
```

Comment:
I guess we didn't need to do this during cleanup...?

Reply:
This got folded/standardized into RestartNestedTransaction